src/game.cpp

```diff
@@ -7745,38 +7745,46 @@ bool pet_menu(monster *z)
         drop_all,
         give_items,
         pheromone,
-        rope
+        rope,
+        inject
     };
 
     uimenu amenu;
 
-    std::string pet_name = _("dog");
-    if( z->type->in_species( ZOMBIE ) ) {
-        pet_name = _("zombie slave");
-    }
+    std::string pet_name = z->name();
 
     amenu.selected = 0;
-    amenu.text = string_format(_("What to do with your %s?"), pet_name.c_str());
+    if (z->has_effect("wary")) {
+        amenu.text = string_format(_("The %s seems to be wary around you."), pet_name.c_str());
+    } else { 
+        amenu.text = string_format(_("What to do with your %s?"), pet_name.c_str()); 
+    }
     amenu.addentry(cancel, true, 'q', _("Cancel"));
 
     amenu.addentry(swap_pos, true, 's', _("Swap positions"));
     amenu.addentry(push_zlave, true, 'p', _("Push %s"), pet_name.c_str());
     amenu.addentry( rename, true, 'e', _("Rename") );
 
-    if (z->has_effect("has_bag")) {
-        amenu.addentry(give_items, true, 'g', _("Place items into bag"));
-        amenu.addentry(drop_all, true, 'd', _("Drop all items"));
-    } else {
-        amenu.addentry(attach_bag, true, 'b', _("Attach bag"));
+    if (!z->has_effect("wary")) {
```

Comment:
You keep checking this effect multiple times. You should cache it as a `bool` (like, `bool wary = z->has_effect("wary");` somewhere above) or group those menu entries that care about it in one `if` block.